opts/options.go

```diff
@@ -177,3 +181,29 @@ func DisableAutoRefresh() Option {
 		return nil
 	}
 }
+
+// EnableProviders enables storing and retrieving provider records.
+//
+// Defaults to true.
+//
+// WARNING: do not change this unless you're using a forked DHT (i.e., a private
+// network and/or distinct DHT protocols with the `Protocols` option).
+func EnableProviders(enable bool) Option {
```

Comment:
nit: Do we prefer this option style, versus a no-arg `DisableProviders()`? (which makes a lot more explicit what the default is, and what the only possible direction of change is (true => false)?

Reply:
See above for `DisableAutoRefresh()`, which uses that style. Let's stick to a single style for DX?

Reply:
I agree we should be consistent. In go-libp2p, we have:

* `EnableRelay(opts)`
* `DisableRelay()`
* `Ping(enabled bool)`
* `NATPortMap()`

Which are all inconsistent.

---

Personally, I prefer the style that takes a bool.

1. Instead of:

```go
opts := ...
if disableThing {
  opts = append(opts, DisableThing())
}
NewThing(opts...)
```

I can write:

```go
NewThing(..., EnableThing(!disableThing))
```

2. I can override defaults. That is, I can append `EnableThing(true)` to a set of options that already includes `EnableThing(false)`.

(although, `WithThing(bool)` may be cleaner)

---

I'll change it to `DisableFoo()` for now but we'll want to make a consistency pass through all of libp2p eventually.

Reply:
I agree we should be consistent. In go-libp2p, we have:

* `EnableRelay(opts)`
* `DisableRelay()`
* `Ping(enabled bool)`
* `NATPortMap()`

Which are all inconsistent.

---

Personally, I prefer the style that takes a bool.

1. Instead of:

```go
opts := ...
if disableThing {
  opts = append(opts, DisableThing())
}
NewThing(opts...)
```

I can write:

```go
NewThing(..., EnableThing(!disableThing))
```

2. I can override defaults. That is, I can append `EnableThing(true)` to a set of options that already includes `EnableThing(false)`.

(although, `WithThing(bool)` may be cleaner)

---

I'll change it to `DisableFoo()` for now but we'll want to make a consistency pass through all of libp2p eventually.